dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/dto/ApiYARNApplicationDTO.java

```diff
@@ -16,146 +16,56 @@
  */
 package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import java.util.List;
-import java.util.Map;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.util.StringUtils;
 
 /**
  * DTO class for the official Cloudera API <a
  * href="https://archive.cloudera.com/cm7/7.0.3/generic/jar/cm_api/apidocs/json_ApiYarnApplication.html">json</a>
+ * with additional information about cluster.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class ApiYARNApplicationDTO {
-  @JsonProperty(required = true)
-  private String clusterName;
-
-  @JsonProperty(required = true)
-  private String applicationId;
-
-  @JsonProperty(required = true)
-  private String name;
-
-  @JsonProperty(required = true)
-  private String state;
-
-  @JsonProperty(required = true)
-  private String startTime;
-
-  @JsonProperty(required = true)
-  private String endTime;
-
-  @JsonProperty(required = true)
-  private String user;
-
-  @JsonProperty(required = true)
-  private String pool;
 
   @JsonProperty(required = true)
-  private List<String> applicationTags;
-
-  @JsonProperty(required = true)
-  private long allocatedMemorySeconds;
-
-  @JsonProperty(required = true)
-  private long allocatedVcoreSeconds;
+  private String clusterName;
 
-  @JsonProperty(required = true)
-  private Map<String, String> attributes;
+  /**
+   * DTO class for the official Cloudera API <a
+   * href="https://archive.cloudera.com/cm7/7.0.3/generic/jar/cm_api/apidocs/json_ApiYarnApplication.html">json</a>.
+   * JsonNode used to support schema evaluation.
+   */
+  private JsonNode apiYARNApplication;
 
-  public String getClusterName() {
-    return clusterName;
+  public ApiYARNApplicationDTO(JsonNode apiYARNApplication) {
+    this.apiYARNApplication = apiYARNApplication;
   }
 
+  @JsonIgnore
   public String getApplicationId() {
-    return applicationId;
-  }
-
-  public String getName() {
-    return name;
-  }
-
-  public String getState() {
-    return state;
-  }
-
-  public String getStartTime() {
-    return startTime;
-  }
-
-  public String getEndTime() {
-    return endTime;
-  }
-
-  public String getUser() {
-    return user;
-  }
-
-  public String getPool() {
-    return pool;
-  }
-
-  public List<String> getApplicationTags() {
-    return applicationTags;
-  }
-
-  public long getAllocatedMemorySeconds() {
-    return allocatedMemorySeconds;
-  }
-
-  public long getAllocatedVcoreSeconds() {
-    return allocatedVcoreSeconds;
+    if (apiYARNApplication == null) {
+      return null;
```

Comment:
Should the response be Optional<String> instead of just null?

Reply:
Hm, is it a known style guide for this?
I saw few examples where it is used for fields as well: https://github.com/spring-projects/spring-boot/blob/0351e33b18e7235fe040ae6b131cd9b1ac951659/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/result/view/MustacheView.java#L118

Reply:
But in these examples it's return type of methods... It's what I say. it's ok.